refactor managers and handlers.

stopfstedt opened the pull request:
shedding some weight for the summer. :bikini:

homu commented:
:umbrella: The latest upstream changes (presumably #1399) made this pull request unmergeable. Please resolve the merge conflicts.

stopfstedt commented:
recap:

- standardized CRUD-method names in managers, then consolidated them as much as possible into `BaseManager` and `DTOManager`. 
- refactored handlers to _use_ managers rather than extending them. then consolidated all handers into one class `BaseHandler` and rewired everything.
- updated all classes that are interfacing with handlers and managers, most notably controllers, voters and commands.
- minor cleanup like code comments corrections and method/class-name case corrections.

stopfstedt commented:
140 files less.

```
~/dev/projects/ilios on master
$ find src -type f | wc -l
991
```

vs. 

```
~/dev/projects/ilios on refactor_handlers
$ find src -type f | wc -l
848
```

jrjohnson commented:
Holding off on merging until post 3.11

jrjohnson commented:
Reviewed as much as possible.  Great stuff, lots of less code 💯 this is ready to go once 3.12 is released because I would like to merge it with plenty of time to test.
